osquery/dispatcher/task_schedule.h

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <functional>
+#include <vector>
+
+#include <osquery/system.h>
+
+namespace osquery {
+
+class TaskSchedule {
+ public:
+  /**
+   * @brief Type alias for a time
+   * TODO: it would be good to use std::chrono::time_point here one day
+   */
+  using UnixTime = decltype(getUnixTime());
+
+  /**
+   * @brief The implementation of task in form of function
+   *
+   * @param scheduled_time The scheduled time of this particular run of task
+   * @return The scheduled time of next run
```

Comment:
Something like `@return The time for the next run or 0 if the task shouldn't be scheduled again` would be more clear.

Reply:
Got it, done